amlip_cpp/test/blackbox/efficiency/workload_distribution/asyncMainComputingNodeTest.cpp

```diff
@@ -0,0 +1,211 @@
+        eprosima::utils::sleep_for(EXECUTION_TIME);
+        return types::JobSolutionDataType(job->to_string());
+    }
+};
+
+eprosima::utils::Duration_ms time_expected(unsigned int n_clients, unsigned int n_servers, unsigned int n_messages)
+{
+    auto total_messages = n_clients * n_messages;
+    if (eprosima::utils::fast_module(total_messages, n_servers) == 0)
+    {
+        return eprosima::utils::fast_division(total_messages, n_servers) * MESSAGES_REPLY_EXPECTED_TIME;
+    }
+    else
+    {
+        return (eprosima::utils::fast_division(total_messages, n_servers) + 1) * MESSAGES_REPLY_EXPECTED_TIME;
+    }
+}
+
+void send_messages(
+    std::vector<std::unique_ptr<node::AsyncMainNode>>& main_nodes, unsigned int n_messages)
+{
+    std::shared_ptr<types::JobDataType> job(std::make_shared<types::JobDataType>(MESSAGE_INTERNAL_TEST));
+    for (auto& main_node : main_nodes)
+    {
+        for (unsigned int i=0; i<n_messages; ++i)
+        {
+            main_node->request_job_solution(job);
+        }
+    }
+}
+
+void wait_solutions(
+    std::vector<std::shared_ptr<test::TestSolutionListener>>& main_listeners, unsigned int n_messages)
+{
+    for (auto& listener : main_listeners)
+    {
+        listener->waiter.wait_equal(n_messages);
+    }
+}
+
+eprosima::utils::Duration_ms execute_nodes(unsigned int n_clients, unsigned int n_servers, unsigned int n_messages)
+{
+    //////
+    // COMPUTING NODES RUNNING
+    std::vector<std::shared_ptr<test::TestTaskListener>> computing_listeners;
+    std::vector<std::unique_ptr<node::AsyncComputingNode>> computing_nodes;
+    for (unsigned int i=0; i<n_servers; i++)
+    {
+        computing_listeners.emplace_back(std::make_shared<test::TestTaskListener>());
+        computing_nodes.emplace_back(
+            std::make_unique<node::AsyncComputingNode>(
+                "TestComputingNode",
+                computing_listeners[i]));
+        computing_nodes[i]->run();
+    }
+
+    //////
+    // MAIN NODES RUNNING
+    std::vector<std::shared_ptr<test::TestSolutionListener>> main_listeners;
+    std::vector<std::unique_ptr<node::AsyncMainNode>> mains_nodes;
+    for (unsigned int i=0; i<n_clients; i++)
+    {
+        main_listeners.emplace_back(std::make_shared<test::TestSolutionListener>());
+        mains_nodes.emplace_back(
+            std::make_unique<node::AsyncMainNode>(
+                "TestMainNode",
+                main_listeners[i]));
+    }
+
+    //////
+    // SEND MESSAGES AND MEASURE TIME
+    eprosima::utils::Timer timer;
+
+    send_messages(mains_nodes, n_messages);
+    wait_solutions(main_listeners, n_messages);
+
+    auto elapsed_time = timer.elapsed_ms();
+
+    return elapsed_time;
+}
+
+void execute_test(unsigned int n_clients, unsigned int n_servers, unsigned int n_messages)
+{
+    auto time_elapsed = execute_nodes(n_clients, n_servers, n_messages);
+    ASSERT_LT(time_elapsed, time_expected(n_clients, n_servers, n_messages) + RESIDUAL_TIME);
```

Comment:
Why equal?

Reply:
because when I was running the tests it gave me fail because it was equal and not less. But then I realised that it also gave me a higher value, so that doesn't work either 🤕